### PR TITLE
Move authentication related methods to FAB auth manager

### PR DESCRIPTION
**Pull request by vincbeck**
One of the many PR which move stuff related to FAB user management from security manager to auth manager.

Code only being moved here.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Rebase needed.

File: airflow/auth/managers/fab/security_manager/override.py [hunk 1]
```diff
@@ -103,6 +112,16 @@ class FabAirflowSecurityManagerOverride(AirflowSecurityManagerV2):
...
 
+    """ Flask-JWT-Extended """
+    jwt_manager = None
+    """ Flask-OpenID OpenID """
+    oid = None
+    """ OAuth email whitelists """
+    oauth = None
+    """ OAuth email whitelists """
+    oauth_remotes: dict[str, Any]
+    oauth_user_info = None
```
Comment (uranusjr) [inline airflow/auth/managers/fab/security_manager/override.py]: I believe the convention is to put docstrings _below_ attributes but I also see there are already other variables doing this…

https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html

Comment (vincbeck) [reply airflow/auth/managers/fab/security_manager/override.py]: My bad, I'll fix it

Comment (vincbeck) [reply airflow/auth/managers/fab/security_manager/override.py]: Done

**Comment by vincbeck** [comment]
> Rebase needed.

Done

**Review by potiuk — APPROVED** [review]

**Review by uranusjr — APPROVED** [review]

File: airflow/auth/managers/fab/security_manager/override.py [hunk 2]
```diff
@@ -1650,3 +1950,82 @@ def _ldap_calculate_user_roles(self, user_attributes: dict[str, list[bytes]]) ->
...
+        role_resource_names = [
+            perm.resource.name for perm in self.filter_roles_by_perm_with_action(action_name, db_role_ids)
+        ]
+        result.update(role_resource_names)
+        return result
+
+    def filter_roles_by_perm_with_action(self, action_name: str, role_ids: list[int]):
+        """Find roles with permission."""
+        return (
+            self.appbuilder.get_session.query(self.permission_model)
```
Comment (uranusjr) [inline airflow/auth/managers/fab/security_manager/override.py:2019]: Should we change this to be compatible with SQLALchemy 2?

Comment (vincbeck) [reply airflow/auth/managers/fab/security_manager/override.py:2019]: Let's do it in another PR. Some other code might not be compatible with SQLALchemy 2, so let's create another PR for that purpose

**Pull request merged**